exporter/awsemfexporter/datapoint.go

```diff
@@ -137,20 +150,19 @@ func (dps numberDataPointSlice) At(i int) (dataPoint, bool) {
 		}
 	}
 
-	return dataPoint{
-		value:       metricVal,
-		labels:      labels,
-		timestampMs: timestampMs,
-	}, retained
+	datapoints = append(datapoints, dataPoint{name: dps.metricName, value: metricVal, labels: labels, timestampMs: timestampMs})
+
+	return datapoints, retained
```

Comment:
```suggestion
	return []dataPoint{{name: dps.metricName, value: metricVal, labels: labels, timestampMs: timestampMs}}, retained
```

If I'm reading this correctly this is the only time `datapoints` is modified after it is initialized to an empty slice.  Instead of using `append()` here I would just return a slice literal containing the one value.  the return above can return an empty slice literal.

Reply:
Its the same reason as below
```
I'm fine with that. The reason why I giving its a variable is to keep consistency between calculateDeltaDatapoints function.
```
However, I'm fine with changing this !